src/video/cameradevice.cpp

```diff
@@ -317,7 +317,17 @@ QVector<QPair<QString, QString>> CameraDevice::getDeviceList()
     if (idesktopFormat)
     {
         if (idesktopFormat->name == QString("x11grab"))
-            devices.push_back(QPair<QString,QString>{"x11grab#:0", QObject::tr("Desktop", "Desktop as a camera input for screen sharing")});
+        {
+            QString dev = "x11grab#";
+            QByteArray display = qgetenv("DISPLAY");
+
+            if (display.isNull())
+                dev += ":0";
+            else
+                dev += display.constData();
```

Comment:
unless `$DISPLAY` is malformed, no.

```console
$ echo $DISPLAY
:1.0
```

As with what we want here, it's `[host]:display[.screen]`.  See https://tronche.com/gui/x/xlib/display/opening.html